doc/design/zarr-publish-1.md

```diff
@@ -0,0 +1,71 @@
+# Publish Dandisets that contain Zarr archives, and support updates to the Zarr archive after publishing the Dandiset
+
+This document describes the current implementation of publishing Dandisets with Zarr archives, desired use cases, and the associated requirements.
+
+## Current implementation
+
+When a blob asset is updated, a new version (i.e. a copy) is uploaded to the S3 bucket.  Zarr archives are too large so multiple copies should not be created.  A Zarr archive is uploaded once and it is updated in place.  This design means that the Zarr archive is immutable once the Dandiset is published, so that the published Dandiset is immutable. Currently, a Dandiset cannot be published if it contains a Zarr asset.  For more details, see the [zarr-support-3 design doc](https://github.com/dandi/dandi-archive/blob/master/doc/design/zarr-support-3.md).
```

Comment:
i would add that there are a two requirements.
- updating/publishing in a dandiset
- adding the asset to another dandiset 
- being able to publish both

for implementation:
- we currently don't have a readonly mode for an asset and any modification of zarr added to another dandiset should fail, unless control is handed over (more complicated).  
- adding should have either a link (readonly) or copy option. on the blob side, we have copy on write, but for assets this could get complicated for large trees.